ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -1499,10 +1500,10 @@ private static ValidTxnList getValidTxnList(Configuration conf) {
   public static Map<Path, HdfsDirSnapshot> getHdfsDirSnapshotsForCleaner(final FileSystem fs, final Path path)
           throws IOException {
     Map<Path, HdfsDirSnapshot> dirToSnapshots = new HashMap<>();
-    Deque<RemoteIterator<FileStatus>> stack = new ArrayDeque<>();
-    stack.push(fs.listStatusIterator(path));
+    Deque<RemoteIterator<LocatedFileStatus>> stack = new ArrayDeque<>();
+    stack.push(FileUtils.listAndFilter (fs, path, true, acidHiddenFileFilter));
```

Comment:
Earlier it was listStatusIterator, you changed it to listFiles with recursive, it won't be having directories now, empty directories and stuff like that will get completely skipped, I am not sure if you can do that. Lets not change the behaviour in this scope.